Chess/game.cpp

```diff
@@ -1,40 +1,166 @@
 #include "game.h"
+#include "desk.h"
+#include "command.h"
+
 #include <qfiledialog.h>
+#include <QVariant>
 
 Game::Game(QObject *parent) :
-    //QObject(parent),
-    viewer_(parent)
+    root_(parent),
+    desk_(NULL),
+    command_(NULL)
 {
     desk_ = new Desk(this);
 }
 
+Cell Game::parseQMLCellName(QString name)
+{
+    //needs err handle here
+    if (!name.contains(QRegExp("cell[1-8][1-8]")))
+        return Cell(-1,-1);

```

Comment:
I would prefere exceprions instead of return codes. 
If you use return codes do not return invalid object, separate returned object from return code.